content/utilities/PlanetOverview/plugin.js

```diff
@@ -0,0 +1,338 @@
+    topSlider.onchange = (evt) => {
+      try {
+        this.topX = parseInt(evt.target.value, 10);
+        dynamicLabel.innerText = `Top ${this.topX} of PlanetType: ${this.planetType} up Lvl: ${this.minLevel}`;
+        this.renderPlanets();
+      } catch (err) {
+        console.error("Unable to parse number", err);
+      }
+    };
+
+    // label for dynamic information
+    const dynamicLabel = document.createElement("label");
+    dynamicLabel.style.width = "50%";
+    dynamicLabel.style.padding = "5px 0";
+    dynamicLabel.style.marginRight = "170px";
+    dynamicLabel.innerText = `Top ${this.topX} of PlanetType: ${this.planetType} up Lvl: ${this.minLevel}`;
+
+    // label for my planets
+    const myPLanetsLabel = document.createElement("label");
+    myPLanetsLabel.innerText = `only my planets?`;
+    myPLanetsLabel.style.marginRight = "20px";
+
+    // Checkbox for my planets ?
+    let MyPlanets = document.createElement("input");
+    MyPlanets.type = "Checkbox";
+    MyPlanets.checked = true;
+    MyPlanets.id = "myPlanets";
+    this.MyPlanets = MyPlanets;
+    MyPlanets.onchange = (evt) => {
+      try {
+        dynamicLabel.innerText = `Top ${this.topX} of PlanetType: ${this.planetType} up Lvl: ${this.minLevel}`;
+        this.renderPlanets();
+      } catch (err) {
+        console.error("Unable to change checkbox", err);
+      }
+    };
+
+    // This is the fuction for button upgrade that show current unconfirmed tx
+    function updateTx() {
+      dynamicLabel.innerText =
+        "Moves :  " +
+        df.getUnconfirmedMoves().length +
+        "  Upgrades :  " +
+        df.getUnconfirmedUpgrades().length +
+        "  Wormhole :  " +
+        df.getUnconfirmedWormholeActivations().length;
+    }
+
+    // This is the function that shows the transactions in the console
+    function unconfirmed() {
+      console.log("Moves : ", df.getUnconfirmedMoves());
+      console.log("Upgrades : ", df.getUnconfirmedUpgrades());
+      console.log("Wormhole : ", df.getUnconfirmedWormholeActivations().length);
+    }
+    // Formating of big numbers silver + energy
+    
+    function roundToDecimal(num, decimalCount = 1) {
+      if (decimalCount < 1) return Math.round(num);
+      let p = Math.pow(10, decimalCount);
+      num = num * p;
+      num = Math.round(num) / p;
+      return num;
+    }
+    
+    function formatNumberForDisplay(num, decimalCount = 1) {
+      if (num < 1e3) return roundToDecimal(num, decimalCount);
+      if (num < 1e6) return roundToDecimal(num / 1e3, decimalCount) + "k";
+      if (num < 1e9) return roundToDecimal(num / 1e6, decimalCount) + "m";
+      if (num < 1e12) return roundToDecimal(num / 1e9, decimalCount) + "b";
+      return roundToDecimal(num / 1e12, decimalCount) + "t";
+    }
+
+    // Grafic append
+    container.appendChild(planetType);
+    container.appendChild(minPlanetLevel);
+    container.appendChild(updateButton);
+    container.appendChild(unconfirmedButton);
```

Comment:
Since this is a debug option maybe just hide it here?
```suggestion
//    container.appendChild(unconfirmedButton);
```

Reply:
I would like to keep, one of my most used function.

Reply:
You can keep it for your plugin, is np, but you are uploading this for other people to use. Other people will not know what this button does and why it is here. You could just easily comment it out for uploading and then uncomment it again for your own use.
I see you said:
`for example if you wanna do refresh and have a list of important moves...`
So you use this button to know when you should refresh the plugin?
You could maybe also add a description to the button to tell people what its good for, for exmaple:
`unconfirmedButton.title = "logs unconfirmed transactions to the console (F12)";`
this would appear if you hover over the button for a short time.